InterviewFlashCards/IFCFlashCard.m

```diff
@@ -86,9 +86,13 @@ - (void)prepareFlashCardWithCompletion:(void (^)())completion{
     } else {
         self.answerImagesLoaded = YES;
         if (self.questionImagesLoaded) {
-            dispatch_async(dispatch_get_main_queue(), ^{
+            if (NSThread.isMainThread) {
                 completion();
-            });
+            } else {
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    completion();
+                });
+            }
```

Comment:
@mesbekmek Good question. So the tests run synchronously, which is how I found this in the first place - because it runs synchronously the changes caused by the completion block wasn't being noticed. One option is to set an expectation and fulfill it, but another way would be to dispatch_async to the main thread only if we aren't already on the main thread. 
When we use `dispatch_async` on a `serial queue` like the `main thread`, the code block gets added to the queue at the end of the current run loop, which is why it is slightly delayed. By not dispatching here, we will just synchronously inline the code block in the current run loop.